### PR TITLE
fixed issue in modify enrollment

### PR DESCRIPTION
**Pull request by hasuniea**

### CODE REVIEW

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java [hunk 1]
```diff
@@ -346,12 +344,6 @@ private boolean enrollDevice(Document request) throws WindowsDeviceEnrolmentExce
                 }
                 Device generateDevice = generateDevice(DeviceManagementConstants.MobileDeviceTypes.
                         MOBILE_DEVICE_TYPE_WINDOWS, devID, modVersion, imsi, imei, devMan, devMod, user);
```
Comment (harshanL) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java]: Can we reduce the number of parameters in the generateDevice method?

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java [hunk 2]
```diff
@@ -346,12 +344,6 @@ private boolean enrollDevice(Document request) throws WindowsDeviceEnrolmentExce
...
-                DeviceIdentifier deviceIdentifier = convertToDeviceIdentifierObject(syncmlHeader.getSource()
-                        .getLocURI());
-               // Device existingDevice = WindowsAPIUtils.getDeviceManagementService().getDevice(deviceIdentifier);
-                if (WindowsAPIUtils.getDeviceManagementService().getDevice(deviceIdentifier) != null) {
-                    WindowsAPIUtils.getDeviceManagementService().modifyEnrollment(generateDevice);
-                }
                 status = WindowsAPIUtils.getDeviceManagementService().enrollDevice(generateDevice);
                 WindowsAPIUtils.startTenantFlow(user);
```
Comment (harshanL) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java]: Why does this method is returning just after staring the tenant-flow?

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java (hunk 1 above)
Comment (hasuniea) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java]: fixed the issue by creating bean class for windowsDevice.
https://github.com/wso2/product-mdm/pull/154

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java (hunk 2 above)
Comment (hasuniea) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java]: already fixed it.Stop the tenant flow in finally block.

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java [hunk 3]
```diff
@@ -323,11 +323,9 @@ private boolean enrollDevice(Document request) throws WindowsDeviceEnrolmentExce
...
         int msgID;
         SyncmlDocument syncmlDocument;
-        SyncmlHeader syncmlHeader;
 
         try {
             syncmlDocument = SyncmlParser.parseSyncmlPayload(request);
-            syncmlHeader = syncmlDocument.getHeader();
             msgID = syncmlDocument.getHeader().getMsgID();
             if (msgID == PluginConstants.SyncML.SYNCML_FIRST_MESSAGE_ID) {
                 Replace replace = syncmlDocument.getBody().getReplace();
```
Comment (DilanUA) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java]: It's inappropriate to use a verb like Replace for a class name, should be some appropriate noun

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java [hunk 4]
```diff
@@ -346,12 +344,6 @@ private boolean enrollDevice(Document request) throws WindowsDeviceEnrolmentExce
                 }
                 Device generateDevice = generateDevice(DeviceManagementConstants.MobileDeviceTypes.
                         MOBILE_DEVICE_TYPE_WINDOWS, devID, modVersion, imsi, imei, devMan, devMod, user);
-                DeviceIdentifier deviceIdentifier = convertToDeviceIdentifierObject(syncmlHeader.getSource()
-                        .getLocURI());
-               // Device existingDevice = WindowsAPIUtils.getDeviceManagementService().getDevice(deviceIdentifier);
-                if (WindowsAPIUtils.getDeviceManagementService().getDevice(deviceIdentifier) != null) {
-                    WindowsAPIUtils.getDeviceManagementService().modifyEnrollment(generateDevice);
-                }
                 status = WindowsAPIUtils.getDeviceManagementService().enrollDevice(generateDevice);
```
Comment (DilanUA) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java]: possible NPE scenario if we do not get the service here. better checking it first and invoke .enrollDevice

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java [hunk 5]
```diff
@@ -346,12 +344,6 @@ private boolean enrollDevice(Document request) throws WindowsDeviceEnrolmentExce
                 }
                 Device generateDevice = generateDevice(DeviceManagementConstants.MobileDeviceTypes.
```
Comment (DilanUA) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java]: generateDevice is not a proper name for a variable. do not include verbs or actions as variable names

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java [hunk 6]
```diff
@@ -369,7 +361,6 @@ private boolean enrollDevice(Document request) throws WindowsDeviceEnrolmentExce
                 DeviceIdentifier deviceIdentifier = convertToDeviceIdentifierObject(syncmlDocument.
                         getHeader().getSource().getLocURI());
                 Device existingDevice = WindowsAPIUtils.getDeviceManagementService().getDevice(deviceIdentifier);
```
Comment (DilanUA) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java]: possible NPE scenario if we do not get the service here. better checking it first and invoke .getDevice

Comment (hasuniea) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java]: already handled the Null check in WindowsAPIUtils.getDeviceManagementService()

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java (hunk 5 above)
Comment (hasuniea) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java]: renamed varible and fixed it from https://github.com/wso2/product-mdm/pull/156

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java (hunk 3 above)
Comment (hasuniea) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java]: already fixed it in earlier PR.

File: modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java (hunk 4 above)
Comment (hasuniea) [inline modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/syncml/impl/SyncmlServiceImpl.java]: WindowsAPIUtils.getDeviceManagementService() method already handled NPE.

**Pull request merged**